mapgen.c

```diff
@@ -1 +1,26 @@
 #include "mapgen.h"
+#define SIZE 20
+
+
+int random_number(int min, int max){
+
+    return (rand()%(max-min+1)+min);
+
+}
+
+void generate_map(char map[100][100],int min,int max){

```

Comment:
Change the arguments to:
void generate_map(struct Gmdt *gmdt)

And at the beginning of this function, generate map size randomly, from whatever range you want, it could be from 15 to 25 for example.
Then make gmdt->rows and gmdt->columns equal to that size.